src/ctapipe/io/interpolation.py

```diff
@@ -0,0 +1,283 @@
+        default_value=True,
+        help="If true, raises an exception when trying to extrapolate out of the given table",
+    ).tag(config=True)
+
+    extrapolate = traits.Bool(
+        help="If bounds_error is False, this flag will specify whether values outside"
+        "the available values are filled with nan (False) or extrapolated (True).",
+        default_value=False,
+    ).tag(config=True)
+
+    table_location = None
+    required_columns = set()
+    expected_units = {}
+
+    def __init__(self, h5file=None, **kwargs):
+        super().__init__(**kwargs)
+
+        if h5file is not None and not isinstance(h5file, tables.File):
+            raise TypeError("h5file must be a tables.File")
+        self.h5file = h5file
+
+        self.interp_options: dict[str, Any] = dict(assume_sorted=True, copy=False)
+        if self.bounds_error:
+            self.interp_options["bounds_error"] = True
+        elif self.extrapolate:
+            self.interp_options["bounds_error"] = False
+            self.interp_options["fill_value"] = "extrapolate"
+        else:
+            self.interp_options["bounds_error"] = False
+            self.interp_options["fill_value"] = np.nan
+
+        self._interpolators = {}
+
+    @abstractmethod
+    def add_table(self, tel_id, input_table):
+        """
+        Add a table to this interpolator
+        This method reads input tables and creates instances of the needed interpolators
+        to be added to _interpolators. The first index of _interpolators needs to be
+        tel_id, the second needs to be the name of the parameter that is to be interpolated
+
+        Parameters
+        ----------
+        tel_id : int
+            Telescope id
+        input_table : astropy.table.Table
+            Table of pointing values, expected columns
+            are always ``time`` as ``Time`` column and other columns for the data
+            that is to be interpolated
+        """
+
+        pass
+
+    def _check_tables(self, input_table):
+        missing = self.required_columns - set(input_table.colnames)
+        if len(missing) > 0:
+            raise ValueError(f"Table is missing required column(s): {missing}")
+        for col in self.expected_units:
+            unit = input_table[col].unit
+            if unit is None or not self.expected_units[col].is_equivalent(unit):
+                raise ValueError(
+                    f"{col} must have units compatible with '{self.expected_units[col].name}'"
+                )
+
+    def _check_interpolators(self, tel_id):
+        if tel_id not in self._interpolators:
+            if self.h5file is not None:
+                self._read_parameter_table(tel_id)  # might need to be removed
+            else:
+                raise KeyError(f"No table available for tel_id {tel_id}")
+
+    def _read_parameter_table(self, tel_id):
+        input_table = read_table(
+            self.h5file,
+            f"{self.table_location}/tel_{tel_id:03d}",
```

Comment:
Somewhat confusing as now `table_location` is not the location of the table, but rather the location of the **group** that contains the set of tables.  Maybe we should use a better name?

Reply:
`telescope_data_group`?